Projects Dashboard

AlexNGerman opened the pull request.

Commit:
Projects Dashboard page(draft)

Commit:
List of Projects
List of Tasks
Add Project
Add Task

Commit:
Remove logs

Commit:
Update styles

Commit:
Projects Boxes structure
Tasks list structure
Delete task mutation
Fix DataTime picker fullWidth

Commit:
Fix test file name for TasksList

VladWithCookies commented on src/features/Home/organisms/ProjectModal/ProjectModal.js:
```diff
@@ -0,0 +1,161 @@
+  const handleClose = () => setOpen(false);
+
+  const [{ fetching, error }, createProject] = useMutation(CREATE_PROJECT);
+  const errorMessage = error && error?.message;
```
simply `error?.message`

VladWithCookies commented on src/features/Home/organisms/Task/Task.js:
```diff
@@ -0,0 +1,60 @@
+
+  const [{fetching}, deleteTask] = useMutation(DELETE_TASK);
+
+  const onClickDeleteTask = (id) => {
```
onDeleteTask

VladWithCookies commented on src/features/Home/organisms/TaskModal/TaskModal.js:
```diff
@@ -0,0 +1,125 @@
+
+const TASK_SCHEMA = Yup.object().shape({
+  content: Yup.string()
+    .min(MIN_LENGTH, `Content be less than ${MIN_LENGTH} characters`)
```
Content can't ...

VladWithCookies commented on src/features/Home/organisms/TaskModal/TaskModal.js:
```diff
@@ -0,0 +1,125 @@
+  const handleClose = () => setOpen(false);
+
+  const [{ fetching, error }, createTask] = useMutation(CREATE_TASK);
+  const errorMessage = error && error?.message;
```
Simply `error?.message`

VladWithCookies commented on src/features/Home/organisms/TaskModal/TaskModal.js:
```diff
@@ -0,0 +1,125 @@
+  const errorMessage = error && error?.message;
+
+  const onSubmit = async (values, { setSubmitting }) => {
+    values.projectId = projectId;
```
Can do it via spread operator instead?

VladWithCookies commented on src/features/Home/organisms/Task/Task.js:
```diff
@@ -0,0 +1,60 @@
+
+const Task = ({ task }) => {
+  const { content, id } = task;
+  const [checked, setChecked] = useState([0]);
```
A little bit strange initial value

VladWithCookies replied:
Not sure if we need this state at all

VladWithCookies commented on src/features/Home/organisms/Task/Task.js:
```diff
@@ -0,0 +1,60 @@
+  }
+
+  const handleToggle = (value) => () => {
+    const currentIndex = checked.indexOf(value);
+    const newChecked = [...checked];
+
+    if (currentIndex === -1) {
+      newChecked.push(value);
+    } else {
+      newChecked.splice(currentIndex, 1);
+    }
+
+    setChecked(newChecked);
```
This can be simplified, i guess. Can we discuss this?

Commit:
Resolve issues after review #14

Commit:
Checkboxes change to updateTask mutation
Tests of ProjectModal

Commit:
Update Datepicker
testid -> find by Placeholder name

Commit:
Update tests

Commit:
Project Page with description
Delete Project button

VladWithCookies commented on src/utils/tests/responses.js:
```diff
@@ -20,3 +18,11 @@ export const signInError = [
+  createProject:{
+    deadline:"2022-02-15T10:36:25.742Z",
+    description:"Test Project - Description",
+    title:"Test Project - Title"
```
Missing space after `:`

VladWithCookies commented on src/mutations/UpdateTask/updateTask.js:
```diff
@@ -0,0 +1,12 @@
+      done
+    }
+  }
+`;
```
Missing final new line

VladWithCookies commented on src/mutations/DeleteProject/deleteProject.js:
```diff
@@ -0,0 +1,9 @@
+      id
+    }
+  }
+`;
```
Missing final new line

VladWithCookies commented on src/features/Home/organisms/Project/Project.js:
```diff
@@ -0,0 +1,34 @@
+            <TasksList tasks={tasks}/>
+          :
+            <p>You don't have any tasks yet</p>
+        }
```
One line?

VladWithCookies commented on src/features/Home/organisms/ProjectModal/__tests__/ProjectModal.test.js:
```diff
@@ -0,0 +1,27 @@
+      userEvent.type(screen.getByPlaceholderText(/Deadline/i), '02/15/2022 12:36 pm')
+      userEvent.click(screen.getByTestId('submit'))
+
+
```
Needless new line

VladWithCookies commented on src/features/Home/organisms/ProjectModal/__tests__/ProjectModal.test.js:
```diff
@@ -0,0 +1,27 @@
+    it('calls handleClose() with correct params', async () => {
+      render();
+
+      userEvent.click(screen.getByTestId('openModal'))
```
Use kebab-case

VladWithCookies commented on src/features/Home/organisms/ProjectModal/__tests__/ProjectModal.test.js:
```diff
@@ -0,0 +1,27 @@
+import renderComponent from 'utils/tests/renderComponent';
+import ProjectModal from 'features/Home/organisms/ProjectModal';
+
+describe('Create Project', () => {
```
ProjectModal

VladWithCookies commented on src/features/Home/organisms/Task/Task.js:
```diff
@@ -0,0 +1,49 @@
+  const { content, id, done } = task;
+  const labelId = `checkbox-list-label-${task}`;
+  const [{fetching}, deleteTask] = useMutation(DELETE_TASK);
+  const [result,updateTask] = useMutation(UPDATE_TASK);
```
Add new line after this line

VladWithCookies replied:
Missing space after `,`

VladWithCookies commented on src/features/Home/organisms/Task/Task.js:
```diff
@@ -0,0 +1,49 @@
+          </IconButton>
+        }
+        disablePadding
+        disabled={fetching || result.fetching}
```
Looks a little bit unclear. Please, provide a better naming for this variables

VladWithCookies commented on src/features/Home/organisms/TaskModal/TaskModal.js:
```diff
@@ -0,0 +1,125 @@
+    .min(MIN_LENGTH, `Content can't be less than ${MIN_LENGTH} characters`)
+    .max(MAX_LENGTH, `Content can't be longer than ${MAX_LENGTH} characters`)
+    .required('Please enter your content')
+
```
Needless new line

VladWithCookies commented on src/features/Home/pages/ProjectPage/ProjectPage.js:
```diff
@@ -0,0 +1,85 @@
+      }
+      {fetching ? <CircularProgress color='primary' />
+                :
+                  (project)
```
useless braces

Commit:
Resolve issues after review #15

Commit:
Move datetime picker to atom component
merge inputProps for fix data-testid

Commit:
Rename testid

Commit:
fix error message

Commit:
Move SubmitButton to global components molecules folder
Fixed DatePicker, added autoComplete

Commit:
Remove empty tests files
circleci config

Commit:
no message

Commit:
Update circleci config

Commit:
Project Modal tests
Task modal tests
remove error alerts in modal
fix warnings on tests signIn
Fix setting Headers on login and fetching data
set timeout for waitFor

Commit:
TaskModal tests set timeout

Commit:
Project Edit:
- Added project header
- Created project edit form & mutation
- added & updated templates

VladWithCookies commented on src/utils/tests/requestHandlers.js:
```diff
@@ -1,5 +1,6 @@
 import { graphql } from 'msw';
 import * as responses from 'utils/tests/responses';
+import {createTask} from "utils/tests/responses";
```
Follow single brace style

VladWithCookies commented on src/components/templates/MainTemplate/MainTemplate.js:
```diff
@@ -3,10 +3,10 @@ import CssBaseline from '@mui/material/CssBaseline';
   <Fragment>
     <CssBaseline />
-    <AppHeader />
+    {header ? header : <AppHeader />}
```
It would be nice to have ability to pass any node as header for template

VladWithCookies commented on src/features/Home/atoms/DeleteProjectButton/DeleteProjectButton.js:
```diff
@@ -0,0 +1,24 @@
+  const [{ fetching }, deleteProject] = useMutation(DELETE_PROJECT);
+
+  const onDeleteProject = async (id) => {
+    const responce = await deleteProject({ id: id });
```
respon**s**e

VladWithCookies commented on src/features/Home/molecules/Project/Project.js:
```diff
@@ -0,0 +1,29 @@
+        <Grid container alignItems="center">
+          <Grid item xs>
+            <Typography gutterBottom variant="h6" component="div">
+              <Link to={`/project/${id}`} title={title} style={{ color: '#1976d2' }}>{title} - ({id})</Link>
```
Why do use inline styles all the time?

VladWithCookies commented on src/features/Home/molecules/Task/Task.js:
```diff
@@ -0,0 +1,41 @@
+const Task = ({ task }) => {
+  const { content, id, done } = task;
+  const labelId = `checkbox-list-label-${task}`;
+  const [{fetching}, updateTask] = useMutation(UPDATE_TASK);
```
Follow consistent brace style

VladWithCookies replied:
Is it fixed?

VladWithCookies commented on src/features/Home/molecules/Task/Task.js:
```diff
@@ -0,0 +1,41 @@
+    <div>
+      <ListItem
+        key={task}
+        secondaryAction={ <DeleteTaskButton id={ id }/>}
```
Needless space

VladWithCookies commented on src/features/Home/molecules/Task/Task.js:
```diff
@@ -0,0 +1,41 @@
+        disablePadding
+        disabled={ fetching }
+      >
+        <ListItemButton onClick={ handleToggle({id: id, done: !done }) } dense>
```
Needless space

VladWithCookies replied:
`onClick` handler looks weird, you have access to the same variables at line 12

VladWithCookies commented on src/features/Home/organisms/AppProjectHeader/AppProjectHeader.js:
```diff
@@ -0,0 +1,24 @@
+import { ROUTES } from 'constants/routes';
+import { Edit, ArrowBack } from "@mui/icons-material";
+
+const AppProjectHeader = ({ project, edit = false }) => {
```
It would be nice to have ability to pass any node instead of predefined strings

VladWithCookies replied:
What about renaming it to ProjectHeader?

VladWithCookies commented on src/utils/tests/responses.js:
```diff
@@ -20,3 +18,17 @@ export const signInError = [
+  createTask: {
+    content: "Test Task Description",
+  }
+};
```
Missing new line

VladWithCookies commented on src/features/Home/pages/ProjectPage/ProjectPage.js:
```diff
@@ -0,0 +1,63 @@
+  return (
+    <MainTemplate header={<AppProjectHeader project={project} />}>
+      {fetching && <CircularProgress color='primary' />}
+      {(project)
```
Needless ()

VladWithCookies commented on src/features/Home/pages/ProjectPage/ProjectPage.js:
```diff
@@ -0,0 +1,63 @@
+  const project = !fetching && data?.project;
+  return (
+    <MainTemplate header={<AppProjectHeader project={project} />}>
+      {fetching && <CircularProgress color='primary' />}
```
Follow single quotation style

VladWithCookies commented on src/features/Home/pages/ProjectEdit/ProjectEdit.js:
```diff
@@ -0,0 +1,118 @@
+  const project = !fetching && data?.project;
+
+  const onSubmit = async (values, { setSubmitting }) => {
+    values = {...values, id }
```
Why not to create new variable instead?

VladWithCookies commented on src/features/Home/pages/ProjectEdit/ProjectEdit.js:
```diff
@@ -0,0 +1,118 @@
+  return (
+    <MainTemplate header={<AppProjectHeader project={project} edit/>}>
+      {fetching && <CircularProgress color='primary' />}
+      {(project)
```
Useless ()

VladWithCookies commented on src/features/Home/pages/Home/Home.js:
```diff
@@ -1,10 +1,40 @@
+                  (projects?.length)
+                    ?
+                      projects.map(project => {
+                        return <Project key={project.createdAt} project={project} />;
```
```projects.map(project => <Project ...```

VladWithCookies commented on src/features/Home/organisms/ProjectModal/__tests__/ProjectModal.test.js:
```diff
@@ -0,0 +1,86 @@
+
+  describe('with valid data', () => {
+    it('calls handleClose() with correct params', async () => {
+
```
Needless new line

Commit:
Resolve issues after review #16

Commit:
Resolve issues after review #16:
- styles

Commit:
Resolve issues after review #16:
- Follow consistent brace style

VladWithCookies commented on package.json:
```diff
@@ -6,14 +6,18 @@
     "graphql": "^16.2.0",
     "history": "^5.2.0",
     "js-cookie": "^3.0.1",
+    "moment": "^2.29.1",
```
Oh, forbidden library )

VladWithCookies commented on src/features/Home/atoms/DatePickerField/DatePickerField.js:
```diff
@@ -0,0 +1,30 @@
+  return (
+    <LocalizationProvider dateAdapter={AdapterDateFns}>
+      <MobileDateTimePicker
+        renderInput={(params) => <TextField {...field} {...props} {...params.inputProps}/>}
```
Missing space before `/>`

VladWithCookies commented on src/features/Home/atoms/DeleteTaskButton/DeleteTaskButton.js:
```diff
@@ -0,0 +1,21 @@
+  }
+
+  return (
+    <IconButton edge='end' aria-label='delete' color='error' onClick={() => onDeleteTask(id)} disabled={ fetching }>
```
Needless spaces around `fetching`

VladWithCookies commented on src/features/Home/molecules/Task/Task.js:
```diff
@@ -0,0 +1,46 @@
+  const [{fetching}, updateTask] = useMutation(UPDATE_TASK);
+
+  const handleToggle = () => {
+    updateTask({id: id, done: !done })
```
Missing space

VladWithCookies commented on src/features/Home/molecules/Task/Task.js:
```diff
@@ -0,0 +1,46 @@
+  return (
+    <ListItem
+      key={task}
+      secondaryAction={ <DeleteTaskButton id={ id }/>}
+      disablePadding
+      disabled={ fetching }
+    ><ListItemButton onClick={ handleToggle } dense>
```
Needless spaces inside `{}`

VladWithCookies commented on src/features/Home/organisms/ProjectHeader/ProjectHeader.js:
```diff
@@ -0,0 +1,24 @@
+    <AppBar position='relative'>
+      <Toolbar>
+        <Typography variant='h6' color='inherit' component='h1' noWrap sx={{ flexGrow: 1 }}>
+          <Button component={Link} to={ edit ? `/project/${id}` : ROUTES.HOME } variant='unbordered' color='white' fontSize='20'>
+            <ArrowBack fontSize='medium' sx={{ mr: 2 }} /> { edit ? `Back to project - "${title}"` : 'Projects List App' }
```
Needless spaces inside `{}`

VladWithCookies commented on src/features/Home/organisms/ProjectHeader/ProjectHeader.js:
```diff
@@ -0,0 +1,24 @@
+import { Link } from 'react-router-dom';
+import { AppBar, Toolbar, Typography, IconButton, Button } from '@mui/material';
+import { ROUTES } from 'constants/routes';
+import { Edit, ArrowBack } from '@mui/icons-material';
```
Should be grouped with other 3rd parties imports

VladWithCookies commented on src/features/Home/organisms/ProjectModal/ProjectModal.js:
```diff
@@ -0,0 +1,144 @@
+import { CREATE_PROJECT } from 'mutations/CreateProject/createProject';
+import { MIN_LENGTH, MAX_LENGTH } from 'constants/auth';
+
+
```
Needless new line

VladWithCookies commented on src/features/Home/organisms/ProjectModal/ProjectModal.js:
```diff
@@ -0,0 +1,144 @@
+                    </Grid>
+                    <Grid item xs={12}>
+                      <DatePickerField
+                        onChange={value => props.setFieldValue('deadline', value)}
```
Can we encapsulate this logic inside `DatePickerField`?

VladWithCookies commented on src/features/Home/organisms/TaskModal/TaskModal.js:
```diff
@@ -0,0 +1,113 @@
+  projectId: '',
+}
+
+const TaskModal = ({projectId}) => {
```
Missing spaces

VladWithCookies commented on src/utils/theme.js:
```diff
@@ -1,3 +1,23 @@
 import { createTheme } from '@mui/material/styles';
-const theme = createTheme();
+const theme = createTheme({
```
Missing new line

Commit:
Resolve issues after review #17:
- reformat code style
- remove moment package
- refactor moment code to data-fns

Commit:
New line

Commit:
Fix id & key

VladWithCookies commented on src/queries/GetProjects/getProjects.js:
```diff
@@ -0,0 +1,20 @@
+import {gql} from 'urql';
```
`src/queries/GetProjects/getProjects.js`
`GetProjects` should be replaced with feature name

VladWithCookies commented on src/queries/GetProject/getProject.js:
```diff
@@ -0,0 +1,21 @@
+import {gql} from 'urql';
```
`src/queries/GetProject/getProject.js`
`GetProject` should be replaced with feature name

VladWithCookies commented on src/mutations/UpdateTask/updateTask.js:
```diff
@@ -0,0 +1,12 @@
+import {gql} from 'urql';
```
`src/queries/UpdateTask/updateTask.js`
`UpdateTask` should be replaced with feature name

VladWithCookies commented on src/mutations/UpdateProject/updateProject.js:
```diff
@@ -0,0 +1,14 @@
+import {gql} from 'urql';
```
`src/queries/UpdateProject/updateProject.js`
UpdateProject should be replaced with feature name

VladWithCookies commented on src/mutations/DeleteTask/deleteTask.js:
```diff
@@ -0,0 +1,9 @@
+import {gql} from 'urql';
```
`src/queries/DeleteTask/deleteTask.js`
'DeleteTask' should be replaced with feature name

VladWithCookies commented on src/mutations/DeleteProject/deleteProject.js:
```diff
@@ -0,0 +1,9 @@
+import {gql} from 'urql';
```
`src/queries/DeleteProject/deleteProject.js`
'DeleteProject' should be replaced with feature name

VladWithCookies commented on src/mutations/CreateTask/createTask.js:
```diff
@@ -0,0 +1,12 @@
+import {gql} from 'urql';
```
`src/queries/CreateTask/createTask.js`
'CreateProject' should be replaced with feature name

VladWithCookies commented on src/mutations/CreateProject/createProject.js:
```diff
@@ -0,0 +1,16 @@
+import {gql} from 'urql';
```
`src/queries/CreateProject/createProject.js`
'CreateProject' should be replaced with feature name

VladWithCookies commented on src/components/molecules/SubmitButton/SubmitButton.js:
```diff
@@ -0,0 +1,14 @@
+    fullWidth
+    variant='contained'
+    disabled={loading}
+  >{loading ? <CircularProgress color='inherit' size={25} /> : children}
```
Missing new line

VladWithCookies commented on src/features/Home/atoms/DatePickerField/DatePickerField.js:
```diff
@@ -0,0 +1,30 @@
+import AdapterDateFns from '@mui/lab/AdapterDateFns';
+import {TextField, FormHelperText} from '@mui/material';
+
+const DatePickerField = ({...props}) => {
```
`(props)`

VladWithCookies commented on src/features/Home/atoms/DatePickerField/DatePickerField.js:
```diff
@@ -0,0 +1,30 @@
+        selected={(field.value && new Date(field.value)) || null}
+        onChange={val => {
+          setFieldValue(field.name, val);
+        }}
```
It would be better to extract this to a variable

VladWithCookies commented on src/features/Home/atoms/DeleteTaskButton/DeleteTaskButton.js:
```diff
@@ -0,0 +1,21 @@
+
+  const onDeleteTask = (id) => {
+    deleteTask({id: id});
+  }
```
```js
  const onDeleteTask = () => {
    deleteTask({id});
  }
  ```

VladWithCookies commented on src/features/Home/molecules/Task/Task.js:
```diff
@@ -0,0 +1,46 @@
+      secondaryAction={<DeleteTaskButton id={id} />}
+      disablePadding
+      disabled={fetching}
+    ><ListItemButton onClick={handleToggle} dense>
```
Add new line

VladWithCookies commented on src/features/Home/organisms/ProjectModal/ProjectModal.js:
```diff
@@ -0,0 +1,139 @@
+        aria-labelledby='project-modal'
+        open={open}
+        data-testid='project-modal'
+      ><DialogTitle sx={{m: 0, p: 2}}>
```
Missing new line

VladWithCookies commented on src/features/Home/organisms/ProjectModal/ProjectModal.js:
```diff
@@ -0,0 +1,139 @@
+              initialValues={INITIAL_VALUES}
+              validationSchema={PROJECT_SCHEMA}
+              onSubmit={onSubmit}
+            >{() => (
```
Missing new line

VladWithCookies commented on src/features/Home/organisms/ProjectModal/__tests__/ProjectModal.test.js:
```diff
@@ -0,0 +1,85 @@
+          expect(screen.getByText(`Title can't be less than 2 characters`)).toBeInTheDocument()
+        });
+        await waitFor(() => {
+          expect(screen.getByText(`Description can't be less than 2 characters`)).toBeInTheDocument()
```
Useless template literal

VladWithCookies commented on src/features/Home/organisms/ProjectModal/__tests__/ProjectModal.test.js:
```diff
@@ -0,0 +1,85 @@
+          expect(screen.getByText(`Title can't be longer than 50 characters`)).toBeInTheDocument()
+        });
+        await waitFor(() => {
+          expect(screen.getByText(`Description can't be longer than 50 characters`)).toBeInTheDocument()
```
Useless template literal

VladWithCookies commented on src/features/Home/organisms/TaskModal/TaskModal.js:
```diff
@@ -0,0 +1,107 @@
+        aria-labelledby='task-modal'
+        open={open}
+        data-testid='task-modal'
+      ><DialogTitle sx={{m: 0, p: 2}}>
```
Missing new line

VladWithCookies commented on src/features/Home/organisms/TaskModal/TaskModal.js:
```diff
@@ -0,0 +1,107 @@
+        open={open}
+        data-testid='task-modal'
+      ><DialogTitle sx={{m: 0, p: 2}}>
+        Add Task - {projectId}
```
Why do we need id in title?

VladWithCookies commented on src/features/Home/organisms/TaskModal/TaskModal.js:
```diff
@@ -0,0 +1,107 @@
+              initialValues={INITIAL_VALUES}
+              validationSchema={TASK_SCHEMA}
+              onSubmit={onSubmit}
+            >{() => (
```
Missing new lline

VladWithCookies commented on src/features/Home/organisms/TaskModal/__tests__/TaskModal.test.js:
```diff
@@ -0,0 +1,67 @@
+        render();
+
+        userEvent.click(screen.getByTestId('open-modal'))
+        userEvent.type(screen.getByTestId('content'), '')
```
Useless line

VladWithCookies commented on src/features/Home/organisms/TaskModal/__tests__/TaskModal.test.js:
```diff
@@ -0,0 +1,67 @@
+        userEvent.click(screen.getByTestId('submit'))
+
+        await waitFor(() => {
+          expect(screen.getByText(`Content can't be less than 2 characters`)).toBeInTheDocument()
```
Needless template literal

VladWithCookies commented on src/features/Home/organisms/TaskModal/__tests__/TaskModal.test.js:
```diff
@@ -0,0 +1,67 @@
+        userEvent.click(screen.getByTestId('submit'))
+
+        await waitFor(() => {
+          expect(screen.getByText(`Content can't be longer than 50 characters`)).toBeInTheDocument()
```
Needless template literal

VladWithCookies commented on src/features/Home/organisms/TaskModal/__tests__/TaskModal.test.js:
```diff
@@ -0,0 +1,67 @@
+        render();
+
+        userEvent.click(screen.getByTestId('open-modal'))
+        userEvent.type(screen.getByTestId('content'), 'content content content content content content content content ')
```
Extra space

VladWithCookies commented on src/features/Home/organisms/TasksList/TasksList.js:
```diff
@@ -0,0 +1,18 @@
+      ?
+      tasks.map(task => {
+        return <Task key={task.createdAt} task={task} />;
+      })
```
Needless return

VladWithCookies commented on src/features/Home/pages/Home/Home.js:
```diff
@@ -1,10 +1,38 @@
+                :
+                  (projects?.length)
+                    ?
+                      projects.map(project => <Project key={project.createdAt} project={project} />)
```
Use id as a key

Commit:
Resolve issues after review #18

Commit:
Move mutations & queries to single folders from sub-folders
